ui/src/containers/Personas/PersonaManage/PersonaAutoComplete/index.js

```diff
@@ -0,0 +1,44 @@
+import React, { PropTypes } from 'react';
+import { compose, setPropTypes } from 'recompose';
+import { fromJS, Map } from 'immutable';
+import ModelAutoComplete from 'ui/containers/ModelAutoComplete';
+
+const enhance = compose(
+  setPropTypes({
+    currentPersonaId: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    selectedPersonaId: PropTypes.string,
+  })
+);
+
+const sort = new Map({ name: 1, _id: 1 });
+
+const searchStringToFilter = (searchString) => {
+  switch (searchString) {
+    case '':
+      return new Map();
+    default:
+      return fromJS({ name: { $regex: searchString, $options: 'i' } });
+  }
+};
+
+// Prevents merging of a persona into itself, etc.
+const removeCurrentPersonaFilter = currentPersonaId => fromJS({
+  _id: { $ne: { $oid: currentPersonaId } }
+});
+
+const getPersonaName = persona => persona.get('name');
+
+const render = ({ selectedPersonaId, currentPersonaId, onChange }) => (
+  <ModelAutoComplete
```

Comment:
When you were showing this yesterday I thought it would have been nice to override the default text shown on the dropdown to "Select person or add new.." - possible via new prop?

Reply:
Taking a look now

Reply:
That is unbelievably deep haha.
- ui/src/components/AutoComplete2/Option/AddNewOption.js
- ui/src/components/AutoComplete2/Options/ModelOptionList/ModelOptionList.js
- ui/src/containers/ModelAutoComplete/index.js
- ui/src/containers/Personas/PersonaManage/PersonaAutoComplete/index.js